platform-dependent-components/problem-solver/cxx/sections-module/utils/sections_utils.cpp

```diff
@@ -27,26 +27,26 @@ ScAddr sections_utils::GetSectionDecompositionTuple(ScMemoryContext * context, S
       ScType::EdgeAccessVarPosPerm,
       SectionsKeynodes::nrel_entity_decomposition);
   ScTemplateSearchResult searchResult;
-  context->HelperSearchTemplate(scTemplate, searchResult);
+  context->SearchByTemplate(scTemplate, searchResult);
   if (!searchResult.IsEmpty())
     return searchResult[0][sections_aliases::DECOMPOSITION_TUPLE];
-  ScTemplateGenResult genResult;
-  ScTemplate::Result result = context->HelperGenTemplate(scTemplate, genResult);
-  if (result)
+  ScTemplateResultItem genResult;
+  context->GenerateByTemplate(scTemplate, genResult);
+
+  if (genResult.Size() != 0)
```

Comment:
not sure about this condition

Reply:
you can just get from genResult by alias
genResult shouldn't be empty if scTemplate isn't empty